cmd/gateway-main.go

```diff
@@ -117,6 +119,31 @@ func newGatewayConfig(accessKey, secretKey, region string) error {
 	return nil
 }
 
+// Return endpoint.
+func parseGatewayEndpoint(arg string) (endPoint string, secure bool, err error) {
+	schemeSpecified, err := regexp.MatchString(".*://", endPoint)
```

Comment:
`arg` instead of `endPoint`? `endPoint` is part of return and at this point it is an empty string.